geneLevel.R

```diff
@@ -172,28 +211,150 @@ get_gene_level_calls <- function(cncf_files,
 #                                                       chrom)))
 #   gene_level[, mean_tcn_per_chr := mean_tcn_per_chr_v[paste(Tumor_Sample_Barcode, chr)]]
 
-  ### annotate integer copy number
-  gene_level <- annotate_integer_copy_number(gene_level, amp_threshold)
-
-  ### remove duplicate entries for partial deletions
-  ### the lower value is chosen on the basis that a
-  ### partial deletion is a deletion but a
-  ### partial amplification is not an amplification
-  gene_level <- gene_level[order(FACETS_CNA, -Nprobes)]
-  gene_level <- gene_level[!duplicated(gene_level, by=c("Tumor_Sample_Barcode", "Hugo_Symbol"))]
-
-  ### Sort & set column classes
-  gene_level[, FACETS_CNA := factor(FACETS_CNA, levels = c(-2:2))]
-  gene_level[, chr := factor(chr, levels = c(1:22, "X", "Y"))]
-  setkey(gene_level, Tumor_Sample_Barcode, Hugo_Symbol)
-  gene_level
+    ### annotate integer copy number
+    gene_level <- annotate_integer_copy_number(gene_level, amp_threshold)
+
+    ### remove duplicate entries for partial deletions
+    ### the lower value is chosen on the basis that a
+    ### partial deletion is a deletion but a
+    ### partial amplification is not an amplification
+    gene_level <- gene_level[order(FACETS_CNA, -Nprobes)]
+    gene_level <- gene_level[!duplicated(gene_level, by=c("Tumor_Sample_Barcode", "Hugo_Symbol"))]
+
+    ### Sort & set column classes
+    gene_level[, FACETS_CNA := factor(FACETS_CNA, levels = c(-2:2))]
+    gene_level[, chr := factor(chr, levels = c(1:22, "X", "Y"))]
+    gene_level = as.data.table(gene_level)
+    setkey(gene_level, Tumor_Sample_Barcode, Hugo_Symbol)
+
+
+    ## Extract purity values
+    ## March 2019, for filters
+    ## within *_hisens.cncf.txt file, the purity is calculated as the maximum value not 1.0 and not NA
+    ## For each unique Tumor_Sample_Barcode
+    ##     subset cf.em; define purity as purity := max(cf.em)
+    gene_level[cf.em < 1, purity:=max(cf.em), by=Tumor_Sample_Barcode]
+    gene_level[, purity:=unique(purity)[!is.na(unique(purity))], by=Tumor_Sample_Barcode]
+    
+    ## define CFcut, the cut-off set for CF
+    ## We suspect 60% of CF is pretty high, so that's the cutoff
+    gene_level[, CFcut := 0.6 * purity]
+
+    ## handle NA values in purity
+    ## if there are NA values in purity, then make CFcut == 10
+    gene_level[, CFcut := ifelse(is.na(CFcut), 10, CFcut)]
```

Comment:
This is actually doing something slightly different. Later in the script we say that 1 way that an AMP can be called an AMP is if the cf.em > CFcut where CFcut = 0.6 * purity. Sometimes the purity is NA so we set the CFcut to 10 (which is arbitrary), which no cf.em would ever reach (since it's bound between 0 and 1) and so the AMP could only be called an AMP if it passes 1 of the other 2 cutoffs (based on genes in the region or tcn.em).

Reply:
Understood. I'll try to reimplement this logic with conditionals. I need this to be readabile/maintainable by future hires.

Reply:
They're setting `CFcut := 10` if purity is `NA` in order for the current conditionals to work. It could be set to 10000 if one wishes, or a different data type, etc.

Reply:
Added 160bc74, where `cf.em>CFcut` is replaced with `(!is.na(purity) & cf.em>CFcut)` resulting in the intended effect of ignoring this filter when purity is NA. Verified that results end up the same.